02-peer-protocol.md

```diff
@@ -1342,6 +1359,47 @@ it's simplest to only allow it to set fee levels; however, as the same
 fee rate applies to HTLC transactions, the receiving node must also
 care about the reasonableness of the fee.
 
+### Signing alternative feerates
+
+When a node sends `commitment_signed`, it may also sign alternative versions
+of the commitment using different feerates.
+
+1. type: 137 (`alternative_commitment_signed`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`u32`:`feerate_per_kw`]
+   * [`signature`:`signature`]
+   * [`u16`:`num_htlcs`]
+   * [`num_htlcs*signature`:`htlc_signature`]
+
+#### Requirements
+
+A sending node:
+  - If `option_alternative_feerates` was negotiated:
+    - MUST send one `alternative_commitment_signed` for each feerate listed
+      in the corresponding `commitment_signed.alternative_feerates`.
+    - MUST use `SIGHASH_ALL` for the `htlc_signature`s.
```

Comment:
>> How does this help with the HTLC txs? You still need to bring your own utxo
>
> No you won't need to, sorry if that was unclear, but the HTLC txs from `alternative_commitment_signed` use `SIGHASH_ALL`, not `SIGHASH_SINGLE | SIGHASH_ANYONECANPAY`, and they do pay the feerate announced.

I see. The fee part was not clear to me from the text. I guess I could have deduced it from using the sighash_all requirement. I think it would be better to explicitly mention this, e.g.:

```suggestion
    - MUST use `SIGHASH_ALL` for the `htlc_signature`s, and apply `feerate_per_kw` for each HTLC transaction even if `option_anchors_zero_fee_htlc_tx` applies to the channel.
```

(to counterbalance bolt-03 being very specific [about the contrary](https://github.com/lightning/bolts/blob/16973e2b857e853308cafd59e42fa830d75b1642/03-transactions.md#requirements-2))

Reply:
Yes definitely, if there is generally a good feeling about the proposal, I'll make that PR much more detailed, since there are quite a few subtleties that are currently hard to grasp.